src/impl/apple/keychain_helper.cpp

```diff
@@ -95,16 +97,32 @@ util::Optional<std::vector<char>> get_key(CFStringRef account, CFStringRef servi
     return std::vector<char>(key_bytes, key_bytes + key_size);
 }
 
-void set_key(const std::vector<char>& key, CFStringRef account, CFStringRef service)
+void set_key(const std::vector<char>& key, CFStringRef account, CFStringRef service, SetKeyMode mode)
```

Comment:
This is starting to look like it'd make more sense as two separate functions, `add_key` and `update_key`, rather than a single function with a mode flag.